addons/mail/static/src/models/thread/thread.js

```diff
@@ -547,8 +547,9 @@ function factory(dependencies) {
         openExpanded() {
             const discuss = this.env.messaging.discuss;
             if (['mail.channel', 'mail.box'].includes(this.model)) {
+                discuss.threadViewer.update({ thread: [['replace', this]] });
                 this.env.bus.trigger('do-action', {
-                    action: 'mail.action_new_discuss',
+                    action: 'mail.action_discuss',
                     options: {
                         clear_breadcrumbs: false,
                         active_id: discuss.threadToActiveId(this),
```

Comment:
Question: this key is useless/counter productive to keep here if thread to open is actually defined by updating the threadViewer?
Are both cases needed sometimes?

Could be worth adding some comments.

Reply:
Most common/logical way of changing/setting thread in discuss is simply by invoking update of thread on thread viewer of discuss. If we could do that 100% of the time, we wouldn't need `active_id` at all...

...But the reason we need `active_id` comes from it's presence in URL: we may want to access to `/web` by directly opening Discuss client action **and** auto-set the thread. `active_id` just does that: humanly formatted/stringified way of referring to a thread, and auto-set thread in discuss on page load.

It still manage things in a complicated manner, by waiting that messaging is initialized **and then** auto-set the thread. This is important because information on channel only comes after messaging initialized, and typos in `active_id` should auto-fallback to mailbox inbox for instance (or at the very least, not create a dummy and wrong thread).

I feel `active_id` is an interesting concept but it's quite poorly implemented right now. Still much better than before our refactoring though, would not change anything right now.